probe/docker/controls.go

```diff
@@ -0,0 +1,70 @@
+package docker
+
+import (
+	"log"
+
+	"github.com/weaveworks/scope/probe/controls"
+	"github.com/weaveworks/scope/report"
+	"github.com/weaveworks/scope/xfer"
+)
+
+// Control IDs used by the docker intergation.
+const (
+	StopContainer    = "docker_stop_container"
+	StartContainer   = "docker_start_container"
+	RestartContainer = "docker_restart_container"
+)
+
+func (r *registry) stopContainer(req xfer.Request) xfer.Response {
+	log.Printf("Got %s", req.Control)
+
+	_, containerID, ok := report.ParseContainerNodeID(req.NodeID)
+	if !ok {
+		return xfer.ResponseErrorf("Invalid ID: %s", req.NodeID)
+	}
+
+	container, ok := r.getContainer(containerID)
+	if !ok {
+		return xfer.ResponseErrorf("Invalid ID: %s", req.NodeID)
+	}
+
+	return xfer.ResponseError(container.Stop())
```

Comment:
I would merge these into:
```Go
return xfer.ResponseError(r.stopContainer(containerID))
```

Which keeps knowledge of the client in the registry.